lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -67,105 +67,111 @@ def fake_run(*args, **kwargs):
 
-        def fake_run(*args, **kwargs):
+        # Record number of times mock functions called by this test
+        functions_called = []
+
+        expected_std_out = b"Some error running tar command, empty tar creation failed"
+
+        def mock_run(*args, **kwargs):
+            functions_called.append("mock_run")
             return subprocess.CompletedProcess(
                 args,
                 returncode=1,
-                stdout=b"Some error running tar command, empty tar creation failed",
+                stdout=expected_std_out,
                 stderr=None,
             )
 
-        monkeypatch.setattr(subprocess, "run", fake_run)
+        monkeypatch.setattr(subprocess, "run", mock_run)
 
-        cp = tool_meister._create_tar(tmp_dir, pathlib.Path(tar_file))
+        cp = tool_meister._create_tar(tmp_dir, Path(tar_file))
         assert cp.returncode == 1
-        assert cp.stdout == b"Some error running tar command, empty tar creation failed"
-        assert (
-            "Tarball creation failed with 1 (stdout 'Some error running tar"
-            f" command, empty tar creation failed') on {tmp_dir}: Re-trying now"
-            in caplog.text
-        )
+        assert cp.stdout == expected_std_out
+        assert functions_called == ["mock_run", "mock_run"]
 
 
 class TestSendDirectory:
-    """Test ToolMeister._send_directory()'s use of ._create_tar()"""
-
-    # Record all the mock functions called by a test
-    functions_called = []
+    """Test ToolMeister._send_directory()"""
 
     directory = tmp_dir / f"{tm_params['hostname']}"
 
     @staticmethod
-    def fake_create_tar(returncode: int, stdout: bytes):
-        def f(directory: pathlib.Path, tar_file: pathlib.Path):
+    def mock_create_tar(returncode: int, stdout: bytes, functions_called: list):
+        def f(directory: Path, tar_file: Path):
+            functions_called.append("mock_create_tar")
             return subprocess.CompletedProcess(
                 args=[], returncode=returncode, stdout=stdout, stderr=None
             )
 
-        __class__.functions_called.append("fake_create_tar")
         return f
 
     @responses.activate
     def test_tar_create_success(self, tool_meister, monkeypatch):
         """This test should pass the tar creation in send directory"""
 
-        def fake_unlink(*args):
-            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
-            self.functions_called.append("fake_unlink")
+        # Record all the mock functions called by this test
+        functions_called = []
+
+        def mock_unlink(*args):
+            assert args[0] == Path(f"{TestSendDirectory.directory}.tar.xz")
+            functions_called.append("mock_unlink")
 
-        def fake_open(*args):
-            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
-            self.functions_called.append("fake_open")
+        def mock_open(*args):
+            assert args[0] == Path(f"{TestSendDirectory.directory}.tar.xz")
```

Comment:
Oddly enough, I believe that we can use `self` in this context to access `directory`.  (E.g., see line 163 in `mock_md5()`. 🙂)  Ditto lines 150 and 199.

Reply:
Yep!